app/controllers/orderproducts_controller.rb

```diff
@@ -67,6 +67,12 @@ def set_orderproduct
       @orderproduct = Orderproduct.find(params[:id])
     end
 
+    def required_user
```

Comment:
what does this line do?

Reply:
It checks whether the user is logged_in or not 
If user not present it redirects you to root_path 
It restricts crud operations for that controller untill the user is logged in as i used before_action